bin/beaker-hostgenerator

```diff
@@ -5,4 +5,8 @@ $LOAD_PATH.unshift(lib_dir)
 
 require 'beaker-hostgenerator'
 
-BeakerHostGenerator::CLI.new.execute!
+cli = BeakerHostGenerator::CLI.new
+
+if not cli then
```

Comment:
This is an alternative to calling `exit` directly within `BeakerHostGenerator::CLI.new` in the circumstance that `--list` is passed as an option on the command line; we return early from the `initialize` method with a `nil` return value. Since otherwise the `initialize` method returns a `BeakerHostGenerator::CLI` object, we can check for a "falsy" value here to determine whether or not to call `execute` (which `nil` does not have) on the returned object.

I'm actually not sure why returning early from the `initialize` method in Ruby causes the `new` method to return the object returned by `initialize` (`nil` in this case). My assumption is that `new` would allocate the object instance then call `initialize` to prepare that object. Maybe @rick or @kevpl could provide some insight here?